MUMUP-2260 : Taco, err um, i mean Bucky Announcements

timlevett opened the pull request:
+ Adds creeper state, :tm:, when you have unread announcements

![image](https://cloud.githubusercontent.com/assets/3534544/11541841/14e9b770-98fa-11e5-964f-616646cb95a9.png)

+ On hover reveals bucky with short title and description

![image](https://cloud.githubusercontent.com/assets/3534544/11541855/236de668-98fa-11e5-9ec4-1196f8a000b1.png)

+ On click on bucky he hides again
+ On click of any header link or the "tell me more" it brings you to features and updates your "last read"
+ On click of "not interested" it just updates your "last read"
+ Multiple Announcement support (up to 3)

![image](https://cloud.githubusercontent.com/assets/3534544/11543226/19058634-9901-11e5-88e3-a20cf469a8f4.png)

_We have 4 new ones here, but we only see the latest 3_

+ Only show announcements that are not expired, if something is expired update it to latest read
+ Mobile friendly mode

![image](https://cloud.githubusercontent.com/assets/3534544/11544513/dcad07b4-9907-11e5-810d-369ca22b1aa1.png)

+ GA this. `category `: feature ; `actions` : hide-bucky, show-more, dismiss; `label` : last id or `na`


### Related changes
+ Refactors popup functionality to work with this. Popup just updates to last popup, announcements update just announcements. Shouldn't see any difference in popup functionality.
+ Move PopupController and partial for modal to the feature section of code. Made more sense there.

keirserrie commented:
Excited to see this :+1:

vertein commented:
:+1:

jhanstra commented:
Awesome work, :+1:

timlevett commented on src/main/webapp/js/app-config.js:
```diff
@@ -3,7 +3,7 @@ define(['angular'], function(angular) {
     config
         .constant('APP_FLAGS', {
-            'features' : false,
+            'features' : true,
```
crap, meant to switch this back to false. brb

keirserrie commented:
Great! Is it possilble to have Bucky go all the way to the top & have the "speech bubble" styling there?

If not, then still :+1: 

![screenshot 2015-12-02 15 38 51](https://cloud.githubusercontent.com/assets/10869628/11545092/f735cab4-990a-11e5-8a1f-06b273b2cc0a.png)

nblair commented:
Bucky winks! This is great!

timlevett commented:
@keirserrie : I can make him go all the way up, forgot about that. The speech bubble might be tough as i'm just doing a border around the whole thing right now.

smargovsky commented:
:+1:  for feature, and :100: for creeper state (:tm:)

smargovsky commented:
@timlevett : Regarding @keirserrie's comment about the arrow - what about using Angular UI Bootstrap popover w/ custom css for the arrow ( like http://www.bootply.com/88325 )

timlevett commented:
Sweet @smargovsky . I'll check it out. Thanks!

timlevett commented:
Took advice from @keirserrie and added in the arrow with help from @smargovsky . Thanks!

![image](https://cloud.githubusercontent.com/assets/3534544/11565185/6d8cfae2-99a2-11e5-8eb0-2852a588bc07.png)

Final product. Note I added notification bell just to make sure all was well when that was enabled.

jhanstra commented:
:+1: again :100:
